modules/store/spec/action_creator.spec.ts

```diff
@@ -0,0 +1,136 @@
+import { createAction, props, union } from '@ngrx/store';
+import { expecter } from 'ts-snippet';
+
+describe('Action Creators', () => {
+  let originalTimeout: number;
+  beforeEach(() => {
+    originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 2000;
+  });
+
+  afterEach(() => {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout;
+  });
+
+  const expectSnippet = expecter(
+    code => `
+  // path goes from root
+  import {createAction, props, union} from './modules/store/src/action_creator';
+    ${code}`,
+    {
+      moduleResolution: 'node',
+      target: 'es2015',
+    }
+  );
+
+  describe('createAction', () => {
+    it('should create an action', () => {
+      const foo = createAction('FOO', (foo: number) => ({ foo }));
```

Comment:
The tests need a line break to separate setup, doing, and assertion.

Reply:
Not sure what you mean here. I added the few line breaks where I think they should be, but the rest mostly look right to me :)

Reply:
I'd prefer

```
      const foo = createAction('FOO', (foo: number) => ({ foo }));

      const fooAction = foo(42);

      expect(fooAction).toEqual({ type: 'FOO', foo: 42 });
```

Not something we specifically need to do now though

Reply:
Done